### PR TITLE
News detail

### PR DESCRIPTION
**Pull request by martinlabate**
### Summary
Implemented News Detail front and back end

### Cards
* https://trello.com/c/PXA7hYBt/12-n-19-frontend-and-backend-news-detail

### Screen Shots
device-2016-08-26-102054
device-2016-08-26-102114
device-2016-08-26-102135
device-2016-08-26-102147
device-2016-08-26-102202
device-2016-08-26-102329

### Gifs
ezgif com-video-to-gif 1

### CODE REVIEW

File: app/src/main/java/com/github/martinlabate/activity/NewsDetailActivity.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+import com.github.martinlabate.Extras;
+import com.github.martinlabate.fragment.NewsDetailFragment;
+
+public class NewsDetailActivity extends SingleFragmentActivity {
+
+    @Override
+    protected Fragment createFragment() {
+        Bundle bundle = new Bundle();
+        bundle
```
Comment (luksdlt92) [inline app/src/main/java/com/github/martinlabate/activity/NewsDetailActivity.java]: Why you did that? The format looks horrible.

File: app/src/main/java/com/github/martinlabate/presenter/NewsPresenter.java [hunk 2]
```diff
@@ -21,11 +23,18 @@
 
     public NewsPresenter(NewsView viewInstance, ParseService parseService) {
         super(viewInstance);
+        EventBus.getDefault().register(viewInstance);
```
Comment (luksdlt92) [inline app/src/main/java/com/github/martinlabate/presenter/NewsPresenter.java]: Why you decided register/unregister EventBus from Presenter and not from Fragment?

Comment (martinlabate) [inline app/src/main/java/com/github/martinlabate/presenter/NewsPresenter.java]: Ema told me it is business logic and therefor should be handled by the presenter.

Comment (luksdlt92) [inline app/src/main/java/com/github/martinlabate/presenter/NewsPresenter.java]: Ok!

File: app/src/main/java/com/github/martinlabate/activity/FullScreenImageActivity.java [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+import com.github.martinlabate.fragment.FullScreenImageFragment;
+
+public class FullScreenImageActivity extends SingleFragmentActivity {
+
+    @Override
+    protected Fragment createFragment() {
+        String imageUri = getIntent().getStringExtra(Extras.FullScreenImage.IMAGE_URI);
+        Bundle bundle = new Bundle();
+        bundle.putString(Extras.FullScreenImage.IMAGE_URI, imageUri);
+        Fragment fullScreenImageFragment = new FullScreenImageFragment();
```
Comment (luksdlt92) [inline app/src/main/java/com/github/martinlabate/activity/FullScreenImageActivity.java]: It's ok for now, but remember create the static method newInstance to create the Bundle.

File: app/src/main/java/com/github/martinlabate/fragment/NewsFragment.java [hunk 4]
```diff
@@ -130,12 +134,31 @@ public void showLoading() {
...
+        }
+    }
+
+    @Subscribe
+    public void onUserToggledLikeEvent(UserToggledLikeNewsEvent event) {
+        int position = 0;
+        while (!mDataset.get(position).getObjectId().equals(event.getNewsId())) {
+            position++;
+        }
+        if (event.isLiked()) {
```
Comment (luksdlt92) [inline app/src/main/java/com/github/martinlabate/fragment/NewsFragment.java]: And if the item is not found it?

File: app/src/main/res/layout/fragmente_news_detail.xml [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+            android:layout_marginLeft="@dimen/spacing_medium_extra"
+            android:layout_marginStart="@dimen/spacing_medium_extra"
+            android:layout_marginEnd="@dimen/spacing_medium_extra"
+            android:layout_marginTop="@dimen/spacing_medium_doble"
+            android:layout_marginBottom="@dimen/spacing_medium_extra"
+            android:orientation="horizontal">
+
+            <TextView
+                android:id="@+id/news_detail_title"
+                android:layout_width="300dp"
```
Comment (luksdlt92) [inline app/src/main/res/layout/fragmente_news_detail.xml]: Hardcoded dimen

File: app/src/main/res/layout/fragmente_news_detail.xml [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+            android:layout_width="match_parent"
+            android:layout_height="@dimen/size_large"
+            android:layout_marginLeft="@dimen/spacing_medium_extra"
+            android:layout_marginRight="@dimen/spacing_medium_extra"
+            android:layout_marginStart="@dimen/spacing_medium_extra"
+            android:layout_marginEnd="@dimen/spacing_medium_extra"
+            android:scrollbars="vertical"
+            android:fillViewport="true">
+
+        <TextView
```
Comment (luksdlt92) [inline app/src/main/res/layout/fragmente_news_detail.xml]: Indentation

**Comment by luksdlt92** [comment]
This is probably your last PR. We would like to congratulate you for the training. We hope you've learned a lot and enjoyed the procress. Welcome to the real world!

:+1: 

rocky-1

**Pull request merged**